msgpack.sh

```diff
@@ -0,0 +1,41 @@
+package: msgpack
+version: v2.1.5
+tag: cpp-2.1.5
+source: https://github.com/msgpack/msgpack-c
+build_requires:
+ - CMake
+ - "GCC-Toolchain:(?!osx)"
+prefer_system: "(?!slc5)"
+prefer_system_check: |
+  printf "#include <msgpack/version.hpp>\n#define VERSION (MSGPACK_VERSION_MAJOR * 10000) + (MSGPACK_VERSION_MINOR * 100) + MSGPACK_VERSION_REVISION\n#if(VERSION < 20105)\n#error \"msgpack version >= 2.1.5 needed\"\n#endif\nint main(){}" | c++ -I$(brew --prefix msgpack)/include -xc++ -std=c++11 - -o /dev/null
```

Comment:
Hi @dennisklein, so - I have tested it on my macOS High Sierra with `msgpack` installed from brew (v3.0.0) and without. Everything's fine with "our" msgpack, but with brew's I get:

```
DEBUG:FairMQ:FairMQ:pr1131: -- FairMQ 1.2.1 from Fri May 4 21:57:00 2018 +0200
DEBUG:FairMQ:FairMQ:pr1131: CMake Error at cmake/FairMQLib.cmake:278 (find_package):
DEBUG:FairMQ:FairMQ:pr1131:   Could not find a configuration file for package "msgpack" that is
DEBUG:FairMQ:FairMQ:pr1131:   compatible with requested version "2.1.5".
DEBUG:FairMQ:FairMQ:pr1131:
DEBUG:FairMQ:FairMQ:pr1131:   The following configuration files were considered but not accepted:
DEBUG:FairMQ:FairMQ:pr1131:
DEBUG:FairMQ:FairMQ:pr1131:     /usr/local/opt/msgpack/lib/cmake/msgpack/msgpack-config.cmake, version: 3.0.0
DEBUG:FairMQ:FairMQ:pr1131:     /usr/local/lib/cmake/msgpack/msgpack-config.cmake, version: 3.0.0
DEBUG:FairMQ:FairMQ:pr1131:
DEBUG:FairMQ:FairMQ:pr1131: Call Stack (most recent call first):
DEBUG:FairMQ:FairMQ:pr1131:   CMakeLists.txt:51 (find_package2)
```

Can you check?

Reply:
So, it looks to me that this is intended behaviour by msgpack. They implemented a version check in their CMake package, that fails on major version change. For now, that means FairMQ `1.2.1` only supports msgpack `2.x`. I will look into it, but this will not be resolved in the context of this PR.

Reply:
FairRootGroup/FairMQ#32